✓ KivyMD. Development - Bounty program

Due to the merger of the KivyMD library version 2.0.0 with the master branch, it is necessary to fix the build of the documentation on Readthedocs. We need to fix the documentation assembly in the 2.0.0 branch - https://github.com/kivymd/KivyMD/tree/2.0.0 and locally (build in a local repository). It is necessary that the documentation build be successful. At the moment, the documentation build fails both locally and on Github :(

The build is currently returning an exception:

```
cd docs
make html
```

Output:

```python
Traceback (most recent call last):
  File "/sphinx/events.py", line 103, in emit
    results.append(listener.handler(self.app, *args))
  File "sphinx/environment/collectors/toctree.py", line 187, in process_doc
    toc = build_toc(doctree)
  File "sphinx/environment/collectors/toctree.py", line 107, in build_toc
    sub_item = build_toc(sectionnode, depth + 1)
  File "sphinx/environment/collectors/toctree.py", line 91, in build_toc
    nodetext = visitor.get_entry_text()
  File "docutils/transforms/parts.py", line 171, in get_entry_text
    return self.get_tree_copy().children
  File "docutils/nodes.py", line 2489, in get_tree_copy
    return self.parent[0]
IndexError: list index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "sphinx/cmd/build.py", line 455, in build_main
    app.build(args.force_all, args.filenames)
  File "sphinx/application.py", line 396, in build
    self.builder.build_update()
  File "phinx/builders/__init__.py", line 365, in build_update
    self.build(
  File "/sphinx/builders/__init__.py", line 387, in build
    updated_docnames = set(self.read())
  File "sphinx/builders/__init__.py", line 503, in read
    self._read_serial(docnames)
  File "sphinx/builders/__init__.py", line 531, in _read_serial
    self.read_doc(docname)
  File "sphinx/builders/__init__.py", line 596, in read_doc
    publisher.publish()
  File "docutils/core.py", line 277, in publish
    self.apply_transforms()
  File "docutils/core.py", line 245, in apply_transforms
    self.document.transformer.apply_transforms()
  File "sphinx/transforms/__init__.py", line 83, in apply_transforms
    super().apply_transforms()
  File "docutils/transforms/__init__.py", line 174, in apply_transforms
    transform.apply(**kwargs)
  File "sphinx/transforms/__init__.py", line 425, in apply
    self.app.emit("doctree-read", self.document)
  File "sphinx/application.py", line 537, in emit
    return self.events.emit(
  File "sphinx/events.py", line 114, in emit
    raise ExtensionError(
sphinx.errors.ExtensionError: Handler <bound method TocTreeCollector.process_doc of <sphinx.environment.collectors.toctree.TocTreeCollector object at 0x104448b20>> for event 'doctree-read' threw an exception (exception: list index out of range)

Extension error (sphinx.environment.collectors.toctree):
Handler <bound method TocTreeCollector.process_doc of <sphinx.environment.collectors.toctree.TocTreeCollector object at 0x104448b20>> for event 'doctree-read' threw an exception (exception: list index out of range)
make: *** [html] Error 2
```

- [x] The problem is at work

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -7,19 +7,19 @@ repos:
 
   # Format Python
   - repo: https://github.com/psf/black
-    rev: 22.3.0
+    rev: 23.12.1
     hooks:
       - id: black
 
   # Sort imports
   - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.13.2
     hooks:
       - id: isort
         additional_dependencies: ["toml"]
 
   # Lint Python
-  - repo: https://gitlab.com/PyCQA/flake8
-    rev: 4.0.1
+  - repo: https://github.com/PyCQA/flake8
+    rev: 7.0.0
     hooks:
       - id: flake8

--- a/docs/sources/_extensions/autoapi_kivymd.py
+++ b/docs/sources/_extensions/autoapi_kivymd.py
@@ -16,11 +16,14 @@ import re
 import autoapi
 import sphinx
 import unidecode
+from autoapi.directives import NestedParse
 from autoapi.extension import LOGGER
 from autoapi.extension import setup as autoapi_setup
 from autoapi.mappers.python.mapper import PythonSphinxMapper
 from autoapi.mappers.python.objects import PythonPythonMapper
+from docutils import nodes
 from sphinx.util.console import bold, darkgreen
+from sphinx.util.nodes import nested_parse_with_titles
 from sphinx.util.osutil import ensuredir
 
 
@@ -128,7 +131,21 @@ def extension_build_finished(app, exception):
                     break
 
 
+def patched_nested_parse_run(self):
+    node = nodes.container()
+    node.document = self.state.document
+    nested_parse_with_titles(self.state, self.content, node)
+    try:
+        title_node = node[0][0]
+        if isinstance(title_node, nodes.title):
+            title_node.children.pop(0)
+    except IndexError:
+        pass
+    return node.children
+
+
 def setup(app):
+    NestedParse.run = patched_nested_parse_run
     PythonPythonMapper.pathname = property(PythonPythonMapper_pathname)
     PythonPythonMapper.include_dir = PythonPythonMapper_include_dir
     PythonSphinxMapper.output_rst = PythonSphinxMapper_output_rst

